fireshape/innerproduct.py

```diff
@@ -352,3 +352,18 @@ def riesz_map(self, v, out):  # dual to primal
         outvec *= 0.
         outvec.setValues(self.global_free_is_col.array, res.array)
         outvec.assemble()
+
+
+class H2InnerProduct(UflInnerProduct):
```

Comment:
This is only justified for wavelets and Bsplines of order > 2. Unless essential, I'd remove it, given that my PhD student is working on making this work for any type of discretization.